src/types.ts

```diff
@@ -10,6 +10,17 @@ export interface Config {
   web3Provider: ethers.providers.Web3Provider;
 }
 
+export interface zNSHub {
+  addRegistrar: (rootDomainId: string, registrar: string) => Promise<void>;
```

Comment:
I feel like this would be a good place to do `tryAddRegistrar` instead in case of failure